src/hyperlight_host/src/sandbox/initialized_multi_use.rs

```diff
@@ -381,6 +381,8 @@ impl MultiUseSandbox {
         ret_type: ReturnType,
         args: Vec<ParameterValue>,
     ) -> Result<ReturnValue> {
+        // Reset snapshot since we are mutating the sandbox state
+        self.snapshot = None;
```

Comment:
Why are we clearing this here?  Shouldn't the restore call above take care of this?

Reply:
otherwise I would have expected it too look like https://github.com/hyperlight-dev/hyperlight/blob/828a529a054ae865efb7a5f480c897bb32cf800d/src/hyperlight_host/src/sandbox/initialized_multi_use.rs#L243

Reply:
I think this is a bug: since this function is wrapping `call_guest_function_by_name_no_reset` and we're mutating the guest sandbox, the attached snapshot will be invalidated here, right? If we don't clear the snapshot, the restore call will hit the "snapshot exists" case and won't do the restore.

Reply:
We could change the behavior to do the `snapshot` and `restore` in the `call_type_erased_guest_function_by_name` though, but since it seemed more clear to do it in the fuzzing code!

Reply:
Oh, I think I miss read this. I think this is fine and mimics the 'call func'

Reply:
Ah ok, I'm glad it looks fine!